go/vt/mysqlctl/replication.go

```diff
@@ -188,6 +189,33 @@ func (mysqld *Mysqld) GetMysqlPort() (int32, error) {
 	return int32(utemp), nil
 }
 
+// GetServerID returns mysql server id
+func (mysqld *Mysqld) GetServerID(ctx context.Context) (uint32, error) {
+	qr, err := mysqld.FetchSuperQuery(ctx, "select @@global.server_id")
+	if err != nil {
```

Comment:
Why is this query here instead of being accessed through the `flavor` interface?

Reply:
This query is the same for all the `flavor` interface implementations. Adding a function there would have meant having the same function for all the implementations. So I have chosen to run this without using the interface. I have followed a similar architecture to `GetMysqlPort` which too tries to run a command that is the same for all the different MySQL - equivalent variants

Reply:
Sounds good.